pkg/services/guardian/accesscontrol_guardian.go

```diff
@@ -282,6 +300,42 @@ func (a *accessControlFolderGuardian) CanView() (bool, error) {
 	return a.evaluate(accesscontrol.EvalPermission(dashboards.ActionFoldersRead, dashboards.ScopeFoldersProvider.GetResourceScopeUID(a.folder.UID)))
 }
 
+func (a *accessControlFolderGuardian) CanTraverse() (bool, error) {
+	if a.folder == nil {
+		return false, ErrGuardianFolderNotFound.Errorf("failed to check permissions for folder")
+	}
+
+	folderPermissions := a.user.Permissions[a.user.OrgID]["dashboards:read"]
```

Comment:
This is a part I rely on. Is it correct that `user.Permissions` always holds effective set of permissions user has, including managed permissions, permissions from teams and roles?